### PR TITLE
Recert to fix test failures with newer Python versions

### PR DESCRIPTION
**Pull request by mgorny**
Newer Python versions have stronger TLS defaults, and therefore reject
the certificates currently used by the test suite.  Regenerate them
to fix most of the test failures.

This also requires bumping the protocol versions used to test min/max
version attributes, as Python no longer allows TLS<1.2.

Unfortunately, test_min_tls_version still fails and I wasn't able to
figure out a way to fix it.

Issue #192

### CODE REVIEW

**Comment by temoto** [comment]
That min_tls_version is giving me nightmares.

**Comment by mgorny** [comment]
Well, I'm open to ideas. For the record, the test failure I get now is:

```
________________________________________________________ test_min_tls_version _________________________________________________________
tests/test_https.py:109: in test_min_tls_version
    http.request(uri)
.tox/pypy3/lib/pypy3.8/site-packages/httplib2/__init__.py:1726: in request
    conn, authority, uri, request_uri, method, body, headers, redirections, cachekey,
.tox/pypy3/lib/pypy3.8/site-packages/httplib2/__init__.py:1441: in _request
    (response, content) = self._conn_request(conn, request_uri, method, body, headers)
.tox/pypy3/lib/pypy3.8/site-packages/httplib2/__init__.py:1363: in _conn_request
    conn.connect()
.tox/pypy3/lib/pypy3.8/site-packages/httplib2/__init__.py:1155: in connect
    self.sock = self._context.wrap_socket(sock, server_hostname=self.host)
/usr/lib/pypy3.8/ssl.py:507: in wrap_socket
    session=session
/usr/lib/pypy3.8/ssl.py:1040: in _create
    self.do_handshake()
/usr/lib/pypy3.8/ssl.py:1309: in do_handshake
    self._sslobj.do_handshake()
/usr/lib/pypy3.8/_cffi_ssl/_stdssl/__init__.py:518: in do_handshake
    raise pyssl_error(self, ret)
E   _cffi_ssl._stdssl.error.SSLError: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure

During handling of the above exception, another exception occurred:
tests/test_https.py:112: in test_min_tls_version
    assert e.reason in ("UNSUPPORTED_PROTOCOL", "VERSION_TOO_LOW")
E   AssertionError: assert 'SSLV3_ALERT_HANDSHAKE_FAILURE' in ('UNSUPPORTED_PROTOCOL', 'VERSION_TOO_LOW')
E    +  where 'SSLV3_ALERT_HANDSHAKE_FAILURE' = SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure').reason
```

**Comment by mgorny** [comment]
(I'm sorry, I've accidentally pasted pypy3 version instead of CPython 3.9 as intended)

**Comment by temoto** [comment]
My best result was either it fails with all combantions of min/max check in tests or it passes without the fix applied. So I thought maybe the correct testing approach should be special Python+openssl build that allows all versions only to find that `x >= min` setting works correctly somewhere. So maybe that test isn't worth troubles and can be deleted. I think that's better than having test that doesn't catch problem.

**Comment by mgorny** [comment]
Hmm, I think you should be actually able to override allowed SSL/TLS versions at runtime. I'm going to try if I can make it work via that.

**Comment by mgorny** [comment]
Nope, sorry, no idea how to make it work. FWICS Python just issues `SSL_CTX_set_min_proto_version()` based on the hardcoded default, so it should be overrideable via setting `minimum_version` and `maximum_version` accordingly but for some reason I seem to get the version error from server init.

**Comment by temoto** [comment]
Yup. We have this code that looks like it calls Python stdlib properly and it's manually verified. Will rely on shoulders of upstream.

**Comment by mgorny** [comment]
Do you want me to update the PR somehow or will you take it from here?

**Comment by temoto** [comment]
@mgorny i found another recert in CI fixing branch, it has additional step to automation in extracting serial. If you don't mind and this version works for you (tests pass on my pc)?

File: tests/__init__.py [hunk 1]
```diff
@@ -725,3 +729,28 @@ def ssl_context(protocol=None):
...
+        return f()(fun)
+
+    # < 3.2
+    return fun
+
+
+@memoize
+def x509_serial(path):
+    if x509 is None:
+        raise Exception("x509_serial requires package cryptography installed")
```
[the remark this answers is not in the document]
Comment (temoto) [reply tests/__init__.py:752]: cryptography was added to requirements-test so will be installed during test

for older versions there is `if x509` condition with fallback to hardcoded cert serial that will go away after dropping py2

[the remark this answers is not in the document]
Comment (temoto) [reply tests/__init__.py:752]: removed test_min_tls_version

**Pull request merged**